dspace-iiif/src/main/java/org/dspace/app/iiif/service/CanvasService.java

```diff
@@ -78,29 +79,51 @@ public CanvasService(ConfigurationService configurationService) {
     }
 
     /**
-     * Checks for bitstream iiif.image.width metadata in the first
-     * bitstream in first IIIF bundle. If bitstream metadata is not
-     * found, use the IIIF image service to update the default canvas
-     * dimensions for this request. Called once for each manifest.
+     * Checks for "iiif.image.width" metadata in IIIF bundles. When bitstream
+     * metadata is not found for the first image in the bundle this method updates the
+     * default canvas dimensions for the request based on the actual image dimensions,
+     * using the IIIF image service. Called once for each manifest.
      * @param bundles IIIF bundles for this item
      */
-    protected void guessCanvasDimensions(List<Bundle> bundles) {
-        Bitstream firstBistream = bundles.get(0).getBitstreams().get(0);
-        if (!utils.hasWidthMetadata(firstBistream)) {
-            int[] imageDims = utils.getImageDimensions(firstBistream);
-            if (imageDims != null && imageDims.length == 2) {
-                // update the fallback dimensions
-                defaultCanvasWidthFallback = imageDims[0];
-                defaultCanvasHeightFallback = imageDims[1];
+    protected void guessCanvasDimensions(List<Bundle> bundles, Context context) {
```

Comment:
The `context` parm is often (not always) the first param in our method... should we try to enforce this rule?

Reply:
Done.